src/helper_functions.jl

```diff
@@ -228,3 +230,122 @@ function check_method_ambiguity_binary(op;
     return result
 end
+
+"""
+    @commutative_neutral(SET, NEUT)
+
+Creates functions to make a set type behave commutative with a given neutral
+element set type.
+
+### Input
+
+- `SET`  -- set type
+- `NEUT` -- set type of the neutral element
+
+### Output
+
+Three function definitions.
+
+### Examples
+
+`@commutative_neutral(ConvexHull, EmptySet)` creates the following functions:
+* `ConvexHull(X::LazySet{N}, ::EmptySet{N}) where {N<:Real} = X`
+* `ConvexHull(::EmptySet{N}, X::LazySet{N}) where {N<:Real} = X`
+* `ConvexHull(Y::EmptySet{N}, ::EmptySet{N}) where {N<:Real} = Y`
+"""
+macro commutative_neutral(SET, NEUT)
+    @eval begin
+        function $SET(X::LazySet{N}, ::$NEUT{N}) where {N<:Real}
+            return X
+        end
+        function $SET(::$NEUT{N}, X::LazySet{N}) where {N<:Real}
+            return X
+        end
+        function $SET(Y::$NEUT{N}, ::$NEUT{N}) where {N<:Real}
+            return Y
+        end
+    end
+    return nothing
+end
+
+"""
+    @commutative_absorbing(SET, ABS)
+
+Creates functions to make a set type behave commutative with a given absorbing
+element set type.
+
+### Input
+
+- `SET`  -- set type
+- `ABS` -- set type of the absorbing element
+
+### Output
+
+Three function definitions.
+
+### Examples
+
+`@commutative_absorbing(MinkowskiSum, EmptySet)` creates the following
+functions:
+* `MinkowskiSum(::LazySet{N}, Y::EmptySet{N}) where {N<:Real} = Y`
+* `MinkowskiSum(Y::EmptySet{N}, ::LazySet{N}) where {N<:Real} = Y`
+* `MinkowskiSum(Y::EmptySet{N}, ::EmptySet{N}) where {N<:Real} = Y`
+"""
+macro commutative_absorbing(SET, ABS)
+    @eval begin
+        function $SET(::LazySet{N}, Y::$ABS{N}) where {N<:Real}
+            return Y
+        end
+        function $SET(Y::$ABS{N}, ::LazySet{N}) where {N<:Real}
+            return Y
+        end
+        function $SET(Y::$ABS{N}, ::$ABS{N}) where {N<:Real}
+            return Y
+        end
+    end
+    return nothing
+end
+
+
+"""
+    @commutative_neutral_absorbing(SET, NEUT, ABS)
+
+Creates functions to make a set type behave commutative with both a given
+neutral and a given absorbing element set type.
+
+### Input
+
+- `SET`  -- set type
+- `NEUT` -- set type of the neutral element
+- `ABS` -- set type of the absorbing element
+
+### Output
+
+Eight function definitions.
+
+### Examples
+
+`@commutative_neutral_absorbing(MinkowskiSum, ZeroSet, EmptySet))` creates the
+following functions:
+* `MinkowskiSum(X::LazySet{N}, ::ZeroSet{N}) where {N<:Real} = X`
+* `MinkowskiSum(::ZeroSet{N}, X::LazySet{N}) where {N<:Real} = X`
+* `MinkowskiSum(Y::ZeroSet{N}, ::ZeroSet{N}) where {N<:Real} = Y`
+* `MinkowskiSum(::LazySet{N}, Y::EmptySet{N}) where {N<:Real} = Y`
+* `MinkowskiSum(Y::EmptySet{N}, ::LazySet{N}) where {N<:Real} = Y`
+* `MinkowskiSum(::ZeroSet{N}, Y::EmptySet{N}) where {N<:Real} = Y`
+* `MinkowskiSum(Y::EmptySet{N}, ::ZeroSet{N}) where {N<:Real} = Y`
+"""
+macro commutative_neutral_absorbing(SET, NEUT, ABS)
```

Comment:
i was wondering if it's possible to work it around such that the extra work done by `commutative_neutral_absorbing` is done on the function `commutative_absorbing`, *provided that* the neutral element of the given set has been defined. we can do this by adding a function `neutral(::set_type)::neutral_type` and a check `isdefined(neutral($SET))`.

Reply:
Hm, we have three of these cases, one with only neutral, one with only absorbing, and one with both neutral and absorbing. So I do not see why `neutral` should play a different role.

But I like the general idea. I would rather add two functions `neutral` and `absorbing` as follows.
The default implementation is `neutral(set) = nothing` and for the respective types we override with `neutral(MinkowskiSum) = ZeroSet`. Then we can get rid of two of the three macros. What would be the name of the remaining macro then?

Reply:
>  Then we can get rid of two of the three macros. 

how is that? i had in mind to substitute the call to `commutative_neutral_absorbing` to the calls `commutative_neutral` and `commutative_absorbing`.  but one wants to do this such that the order of the calls doesn't matter :/

Reply:
You just call this new function (call it `fun`) and internally it calls the `neutral` and `absorbing` functions to see what it should do. Does it make sense?

Reply:
I understood now that what you proposed was very similar, sorry for the confusion.

Reply:
with only 1 macro, the name `commutative_neutral_absorbing` is not the most intuitive IMO since `EmptySet` has nothing to do with `_neutral_` of `CartesianProduct`, while we have to write:

```julia
# EmptySet is the absorbing element for CartesianProduct
absorbing(::Type{CartesianProduct}) = EmptySet

# create absorbing element functions
@commutative_neutral_absorbing(CartesianProduct)
```

idea:

```julia
# EmptySet is the absorbing element for CartesianProduct
@absorbing(CartesianProduct) = EmptySet
```
(or statement like: `Emptyset @absorbs CartesianProduct` or `@is_absorbing_for`)

that unrolls to `absorbing(::Type{CartesianProduct}) = EmptySet` and the function definitions macro. but for a case like M-sum one does:

```julia
@absorbing(MinkowskiSum) = EmptySet
@neutral(MinkowskiSum) = ZeroSet
```

but should also work:

```julia
@neutral(MinkowskiSum) = ZeroSet
@absorbing(MinkowskiSum) = EmptySet
```

Reply:
I admit that this is a very nice solution from the reader's perspective. Note that we will not have documentation for those generated functions, but they are anyway not interesting. Let me see if it works out 👍

Reply:
> `@absorbing(MinkowskiSum) = EmptySet`

I think you cannot do assignments in macros. So I will introduce `@absorbing(MinkowskiSum, EmptySet)`